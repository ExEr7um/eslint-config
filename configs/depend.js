import * as depend from "eslint-plugin-depend"

/** @type {import('eslint').Linter.FlatConfig} */
export default [
  depend.configs["flat/recommended"],

  {
    files: ["package.json"],
    name: "depend/package-json",
    rules: {
      "depend/ban-dependencies": "error",
    },
  },
]
