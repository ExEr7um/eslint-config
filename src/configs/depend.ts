import type { Linter } from "eslint"

import * as depend from "eslint-plugin-depend"

export default [
  depend.configs["flat/recommended"],

  {
    files: ["package.json"],
    name: "depend/package-json",
    rules: { "depend/ban-dependencies": "error" },
  },
] as const satisfies Linter.Config[]
