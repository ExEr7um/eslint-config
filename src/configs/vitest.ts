import type { Linter } from "eslint"

import vitest from "@vitest/eslint-plugin"

export default [
  {
    files: ["**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts}"],
    name: "vitest/base",
    plugins: { vitest },
    rules: {
      ...vitest.configs.recommended.rules,
      "vitest/consistent-test-filename": [
        "error",
        { pattern: String.raw`.*\.spec\.ts` },
      ],
      "vitest/consistent-test-it": ["error", { fn: "test" }],
      "vitest/no-alias-methods": "error",
      "vitest/no-conditional-expect": "error",
      "vitest/no-conditional-in-test": "error",
      "vitest/no-conditional-tests": "error",
      "vitest/no-disabled-tests": "error",
      "vitest/no-focused-tests": ["error", { fixable: false }],
      "vitest/no-interpolation-in-snapshots": "error",
      "vitest/prefer-comparison-matcher": "error",
      "vitest/prefer-each": "error",
      "vitest/prefer-equality-matcher": "error",
      "vitest/prefer-hooks-in-order": "error",
      "vitest/prefer-hooks-on-top": "error",
      "vitest/prefer-strict-equal": "error",
      "vitest/prefer-to-be": "error",
      "vitest/prefer-to-be-falsy": "error",
      "vitest/prefer-to-be-object": "error",
      "vitest/prefer-to-be-truthy": "error",
      "vitest/prefer-to-contain": "error",
      "vitest/prefer-to-have-length": "error",
      "vitest/require-top-level-describe": "error",
    },
  },
] as const satisfies Linter.Config[]
