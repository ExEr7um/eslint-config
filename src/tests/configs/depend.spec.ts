import { describe, expect, test } from "vitest"

import createESLintConfig from "../.."
import depend from "../../configs/depend"

describe("Конфиг depend", () => {
  test("Включен по умолчанию", async () => {
    const config = await createESLintConfig()

    expect(config).toStrictEqual(expect.arrayContaining(depend))
  })

  test("Отключается через параметры", async () => {
    const config = await createESLintConfig({ plugins: { depend: false } })

    expect(config).not.toStrictEqual(expect.arrayContaining(depend))
  })

  test("Имеет все параметры", () => {
    expect(depend).toMatchSnapshot()
  })
})
