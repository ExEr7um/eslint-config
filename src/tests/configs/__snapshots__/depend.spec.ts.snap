// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Конфиг depend > Имеет все параметры 1`] = `
[
  {
    "plugins": {
      "depend": {
        "configs": {
          "flat/recommended": [Circular],
          "recommended": {
            "plugins": [
              "depend",
            ],
            "rules": {
              "depend/ban-dependencies": "error",
            },
          },
        },
        "meta": {
          "name": "eslint-plugin-depend",
          "version": "1.2.0",
        },
        "rules": {
          "ban-dependencies": {
            "create": [Function],
            "meta": {
              "docs": {
                "description": "Bans a list of dependencies from being used",
                "url": "https://github.com/es-tooling/eslint-plugin-depend/blob/main/docs/rules/ban-dependencies.md",
              },
              "messages": {
                "documentedReplacement": ""{{name}}" should be replaced with an alternative package. Read more here: {{url}}",
                "nativeReplacement": ""{{name}}" should be replaced with native functionality. You can instead use {{replacement}}. Read more here: {{url}}",
                "noneReplacement": ""{{name}}" is a banned dependency. An alternative should be used.",
                "simpleReplacement": ""{{name}}" should be replaced with inline/local logic.{{replacement}}",
              },
              "schema": [
                {
                  "additionalProperties": false,
                  "properties": {
                    "allowed": {
                      "items": {
                        "type": "string",
                      },
                      "type": "array",
                    },
                    "modules": {
                      "items": {
                        "type": "string",
                      },
                      "type": "array",
                    },
                    "presets": {
                      "items": {
                        "type": "string",
                      },
                      "type": "array",
                    },
                  },
                  "type": "object",
                },
              ],
              "type": "suggestion",
            },
          },
        },
      },
    },
    "rules": {
      "depend/ban-dependencies": "error",
    },
  },
  {
    "files": [
      "package.json",
    ],
    "name": "depend/package-json",
    "rules": {
      "depend/ban-dependencies": "error",
    },
  },
]
`;
